### PR TITLE
common: STM32 portability enhancement

### PR DESCRIPTION
**Pull request by hermet**
Some systems such as micro-processor might not support the thread feature on the system.

Enhance the portability by compiling the thorvg with toggling the threading depepdency through the build option.

For this, thorvg newly introduced the internal Key/ScopedLock abstraction for transparent thread-locking dependnecy.

To turn off the thread feature, please use the next build option:

$meson setup build -Dthreads=false ...

Note that the thread feature is enabled in default.
Turning off the thread feature could reduce the binary size by 7kb.

issue: https://github.com/thorvg/thorvg/issues/1900

### CODE REVIEW

**Comment by kisvegabor** [comment]
Hi,

Thank you for the prompt action! :100: 

If we are using a managed build can we manually add `#define THORVG_THREAD_SUPPORT 0` to `config.h`?

**Comment by hermet** [comment]
Turning off the thread feature could reduce the binary size by 7kb.

> #define THORVG_THREAD_SUPPORT 0

@kisvegabor Hi, I think you can do that. Alternatively, you can just ignore the option declaration.

File: src/renderer/tvgTaskScheduler.h [hunk 1]
```diff
@@ -86,7 +77,36 @@ struct Task
...
+private:
+    friend struct TaskSchedulerImpl;
+};
+
+#endif  //THORVG_THREAD_SUPPORT
+
+
+struct TaskScheduler
+{
+    static unsigned threads();
```
Comment (kisvegabor) [inline src/renderer/tvgTaskScheduler.h:102]:
```suggestion
    static uint32_t threads();
```

**Review by kisvegabor — COMMENTED** [review]
I've tested it on STM32 and it works, except the minor return type mismatch.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
By the way, how could you request review from me? At LVGL I can do it only for people with write access.

**Comment by hermet** [comment]
@kisvegabor Well... If I'm correct, all registered project members(at least above the triage) on github have permissions to review/approve PRs though only the members with write access could allow merging them.

**Comment by kisvegabor** [comment]
Ah I see. I've tried to check my role, but couldn't find it anywhere. I assume it's "Triage" but seemingly it's not high enough to check my own role :smile: 
Thank you anyway!

**Pull request merged**